api/dataset_test.go

```diff
@@ -686,6 +701,33 @@ func TestPostDatasetReturnsError(t *testing.T) {
 		api := routes(host, secretKey, mux.NewRouter(), store.DataStore{Backend: mockedDataStore})
 		api.router.ServeHTTP(w, r)
 		So(w.Code, ShouldEqual, http.StatusUnauthorized)
+		So(len(mockedDataStore.GetDatasetCalls()), ShouldEqual, 0)
+		So(len(mockedDataStore.UpsertDatasetCalls()), ShouldEqual, 0)
+	})
+
+	Convey("When the dataset already exists and a request is sent to create the same dataset", t, func() {
+		var b string
+		b = datasetPayload
+		r := httptest.NewRequest("POST", "http://localhost:22000/datasets/123", bytes.NewBufferString(b))
+		r.Header.Add("internal-token", secretKey)
+		w := httptest.NewRecorder()
+		mockedDataStore := &storetest.StorerMock{
+			GetDatasetFunc: func(string) (*models.DatasetUpdate, error) {
+				return &models.DatasetUpdate{
+					ID:      "123",
+					Next:    &models.Dataset{},
+					Current: &models.Dataset{},
+				}, nil
+			},
+			UpsertDatasetFunc: func(string, *models.DatasetUpdate) error {
+				return nil
+			},
+		}
+
+		api := routes(host, secretKey, mux.NewRouter(), store.DataStore{Backend: mockedDataStore})
```

Comment:
Then ? 😄

Reply:
This will be technical debt if we go down this route, the BDD standard was originally for the acceptance(integration) tests. Im not sure it should be applied here as unit tests are not driven by business needs necessarily.